substrate/frame/collective/src/lib.rs

```diff
@@ -173,6 +173,138 @@ pub struct Votes<AccountId, BlockNumber> {
+	impl<Deposit, Balance> Convert<u32, Balance> for Constant<Deposit>
+	where
+		Deposit: Get<Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(_: u32) -> Balance {
+			Deposit::get()
+		}
+	}
+
+	/// Linear increasing with some offset.
+	/// f(x) = ax + b, a = `Slope`, x = `proposal_count`, b = `Offset`.
+	pub struct Linear<Slope, Offset>(PhantomData<(Slope, Offset)>);
+	impl<Slope, Offset, Balance> Convert<u32, Balance> for Linear<Slope, Offset>
+	where
+		Slope: Get<u32>,
+		Offset: Get<Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(proposal_count: u32) -> Balance {
+			let base: Balance = Slope::get().saturating_mul(proposal_count).into();
+			Offset::get().saturating_add(base)
+		}
+	}
+
+	/// Geometrically increasing.
+	/// f(x) = a * r^x, a = `Base`, x = `proposal_count`, r = `Ratio`.
+	pub struct Geometric<Ratio, Base>(PhantomData<(Ratio, Base)>);
+	impl<Ratio, Base, Balance> Convert<u32, Balance> for Geometric<Ratio, Base>
+	where
+		Ratio: Get<FixedU128>,
+		Base: Get<Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(proposal_count: u32) -> Balance {
+			Ratio::get()
+				.saturating_pow(proposal_count as usize)
+				.saturating_mul_int(Base::get())
+		}
+	}
+
+	/// Rounds a `Deposit` result with `Precision`.
+	/// Particularly useful for types like [`Geometric`] that might produce deposits with high
+	/// precision.
+	pub struct Round<Precision, Deposit>(PhantomData<(Precision, Deposit)>);
+	impl<Precision, Deposit, Balance> Convert<u32, Balance> for Round<Precision, Deposit>
+	where
+		Precision: Get<u32>,
+		Deposit: Convert<u32, Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(proposal_count: u32) -> Balance {
+			let deposit = Deposit::convert(proposal_count);
+			if !deposit.is_zero() {
+				let factor: Balance = 10u32.pow(Precision::get()).into();
+				(deposit / factor) * factor
+			} else {
+				deposit
+			}
+		}
+	}
+
+	/// Defines `Period` for supplied `Step` implementing [`Convert`] trait.
+	pub struct Stepped<Period, Step>(PhantomData<(Period, Step)>);
+	impl<Period, Step, Balance> Convert<u32, Balance> for Stepped<Period, Step>
+	where
+		Period: Get<u32>,
+		Step: Convert<u32, Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(proposal_count: u32) -> Balance {
+			let step_num = proposal_count / Period::get();
+			Step::convert(step_num)
+		}
+	}
+
+	/// Defines `Delay` for supplied `Step` implementing [`Convert`] trait.
+	pub struct Delayed<Delay, Deposit>(PhantomData<(Delay, Deposit)>);
+	impl<Delay, Deposit, Balance> Convert<u32, Balance> for Delayed<Delay, Deposit>
+	where
+		Delay: Get<u32>,
+		Deposit: Convert<u32, Balance>,
+		Balance: frame_support::traits::tokens::Balance,
+	{
+		fn convert(proposal_count: u32) -> Balance {
+			let delay = Delay::get();
+			if delay > proposal_count {
+				return Balance::zero();
```

Comment:
Why is it returning zero here instead of `Deposit::convert(zero)`? There could be an offset?!